### ISSUE TITLE
@SuppressWarnings does not suppress missing Javadoc on inner class

### ISSUE THREAD

**Issue by WonderCsabo** [checkstyle/checkstyle#2250]
I have this class:

```java
/**
 * This is an example class.
 */
public class BackgroundExecutor {

    @SuppressWarnings("checkstyle:javadoctype")
    public static abstract class Task {
    }
}
```
and this rule file:

```xml
<?xml version="1.0" encoding="UTF-8"?>

<!DOCTYPE module PUBLIC "-//Puppy Crawl//DTD Check Configuration 1.3//EN" "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">

<module name="Checker">
	<module name="TreeWalker">
		<module name="SuppressWarningsHolder" />
		<module name="JavadocType" />
	</module>
	<module name="SuppressWarningsFilter" />
</module>
```

When i run CS i get this output:

```
$ java -jar checkstyle-6.11-all.jar -c checkstyle-checks.xml BackgroundExecutor.java 
Starting audit...
BackgroundExecutor.java:6: error: Missing a Javadoc comment.
Audit done.
Checkstyle ends with 1 errors.
```

But this is incorrect, since the `javadoctype` violation is suppressed by the annotation.
